File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
import java.util.*;

class Solution {
    public int solution(int left, int right) {
        int answer = 0;
        List<Integer> count_list = new ArrayList<>();
        for(int i = left; i <= right; i++){
            int c = count(i);
            count_list.add(c);
        }
        
        for(int i = 0; i < count_list.size(); i++){
            if(count_list.get(i) % 2 != 0){
                answer -= i + left;
            }else{
                answer += i + left;
            }
        }
        
        return answer;
    }
    
    public static int count(int num) {
        int count = 0;
        for(int i = 1; i <= num; i++){
            if(num % i == 0){
                count++;
            }
        }
        return count;
    }
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.swift
import Foundation

func solution(_ left:Int, _ right:Int) -> Int {
    var result = 0
    var arr = Array<Int>()
    for i in left ... right {
        let count = divisor(i)
        arr.append(count)
    }
    
    for i in 0 ..< arr.count {
        if arr[i] % 2 == 0 {
            result += i + left
        }else {
            result -= i + left
        }
    }
    
    
    return result
}

func divisor(_ num: Int) -> Int {
    var count: Int = 0
    for i in 1 ... num {
        if num % i == 0 {
            count += 1
        }
    }
    return count
}